src/osd/PrimaryLogPG.cc

```diff
@@ -769,6 +790,20 @@ class PGLSParentFilter : public PGLSFilter {
                       bufferlist& outdata) override;
 };
 
+bool PGLSNameFilter::filter(const hobject_t &obj,
+                              bufferlist& indata, bufferlist& outdata)
+{
+  // match user regex
+  if (regex != "" && !regex.empty() && regex != "no_reg") {
+    std::string obj_name = obj.oid.name;
+    std::regex e(regex);
```

Comment:
That also should go into init() function. Construction of regex object includes parsig and compiling an regex, and these aren't free.